docs/root/faq/performance/how_to_benchmark_envoy.rst

```diff
@@ -0,0 +1,41 @@
+What are best practices for benchmarking Envoy?
+===============================================
+
+There is :ref:`no single QPS, latency or throughput overhead <faq_how_fast_is_envoy>` that can
+characterize a network proxy such as Envoy. Instead, any measurements need to be contextually aware,
+ensuring an apples-to-apples comparison with other systems by configuring and load testing Envoy
+appropriately. As a result, we can't provide a canonical benchmark configuration, but instead offer
+the following guidance:
+
+* A release Envoy binary should be used. If building, please ensure that `-c opt`
+  is used on the Bazel command line. When consuming Envoy point releases, make
+  sure you are using the latest point release; given the pace of Envoy development
+  it's not reasonable to pick older versions when making a statement about Envoy
+  performance. Similarly, if working on a master build, please perform due diligence
+  and ensure no regressions or performance improvements have landed proximal to your
+  benchmark work and that your are close to HEAD.
+
+* The :option:`--concurrency` Envoy CLI flag should be unset (providing one worker thread per
+  logical core on your machine) or set to match the number of cores/threads made available to other
+  network proxies in your comparison.
+
+* Disable :ref:`generate_request_id
+  <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.generate_request_id>`.
+
+* Disable :ref:`dynamic_stats
+  <envoy_v3_api_field_extensions.filters.http.router.v3.Router.dynamic_stats>`.
+
+* Ensure that the networking and HTTP filter chains are reflective of comparable features
+  in the systems that Envoy is being compared with.
+
+* Ensure that TLS settings (if any) are realistic and that consistent cyphers are used in
+  any comparison.
+
+* Ensure that :ref:`HTTP/2 settings <envoy_v3_api_msg_config.core.v3.Http2ProtocolOptions>`, in
```

Comment:
this reminds me: should we discuss the role of network latency in a benchmark? I would imagine some might want to benchmark performance of a high-latency connection when adding a proxy. In that scenario, it would be interesting to try putting proxy close to upstreams or close to downstreams.